catalog/transformer/catalog-transformer-video-input/src/main/java/ddf/catalog/transformer/input/video/VideoInputTransformer.java

```diff
@@ -0,0 +1,83 @@
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input.video;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.sax.ToXMLContentHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.transformer.common.tika.MetacardCreator;
+import ddf.catalog.transformer.common.tika.TikaMetadataExtractor;
+import net.sf.saxon.TransformerFactoryImpl;
+
+public class VideoInputTransformer implements InputTransformer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VideoInputTransformer.class);
+
+    private static final TransformerFactory TRANSFORMER_FACTORY = new TransformerFactoryImpl();
+
+    private static final String XSLT = "/metadata.xslt";
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        Parser parser = new AutoDetectParser();
+        ToXMLContentHandler handler = new ToXMLContentHandler();
+        TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser, handler);
+
+        Metadata metadata = tikaMetadataExtractor.parseMetadata(input, new ParseContext());
+        String metadataXml = transformToXml(handler.toString());
+
+        return MetacardCreator.createBasicMetacard(metadata, id, metadataXml);
+    }
+
+    private String transformToXml(String xhtml) {
+        LOGGER.debug("Transforming xhtml to xml.");
+        Writer xml = new StringWriter();
+        try {
+            Transformer transformer =
```

Comment:
I believe this transformer can be created once at object construction rather than repeatedly on every transformation.